src/main/java/org/jabref/logic/net/URLDownload.java

```diff
@@ -316,6 +319,7 @@ private void copy(InputStream in, Writer out, Charset encoding) throws IOExcepti
 
     private URLConnection openConnection() throws IOException {
         URLConnection connection = this.source.openConnection();
+        connection.setConnectTimeout((int) connectTimeout.getSeconds() * 1000);
```

Comment:
what about `connectionTimeout.toMillis()`?

Reply:
Good catch. Quite convenient and overseen, thank you.